theseus/embodied/kinematics/robot_model.py

```diff
@@ -39,3 +41,27 @@ def forward_kinematics(self, robot_pose: LieGroup) -> torch.Tensor:
 
     def dim(self) -> int:
         return 1
+
+class UrdfRobotModel(RobotModel):
+    def __init__(self, urdf_path: str):
+        self.robot_model = drm.DifferentiableRobotModel(urdf_path)
+    
+    def forward_kinematics(self, joint_states: torch.Tensor) -> Dict[str, SE3]:
```

Comment:
We should consolidate the API of all `forward_kinematics` classes to match the parent class. To avoid using dicts when it's not practical (e.g., `IdentityModel`), we could maybe define a new type, for example

```python
RobotModelOutput = Union[LieGroup, Dict[str, LieGroup]]
```
which would be used as the output `RobotModel.forward_kinematics`, and let subclasses override with one of the subtypes. Alternatively, we could just pick one data type and use this for all subclasses (i.e., always a dict, or tuple, etc.).

Any thoughts?

Reply:
Suggested interface for RobotModel.forward_kinematics:
```py
forward_kinematics(Union[torch.Tensor, theseus.Vector]) -> RobotModelOutput
```

Will confirm after constructing a working example of an optimization problem using the robot model.

Reply:
Sounds good. I'll try to have a PR ready today with the old code adapted to this change. I'll start with `RobotModelInput` type which I'll set to `Union[torch.Tensor, theseus.Vector]`, and we can revisit this later if this seems overkill.